### PR TITLE
More robust application of access permissions

### PR DESCRIPTION
**Pull request by Alkarex**
We were in particular missing directory search `+X` in our current recommendations.
Extracted to own shell script `cli/access-permissions.sh` so it can easily be invoked.
Update access permissions in Docker to account to be more robust.
#fix https://github.com/FreshRSS/FreshRSS/discussions/5037

### CODE REVIEW

File: Docker/entrypoint.sh [hunk 1]
```diff
@@ -24,6 +22,10 @@ if [ -n "$CRON_MIN" ]; then
 		-r "s#^[^ ]+ #$CRON_MIN #" | crontab -
 fi
 
+./cli/access-permissions.sh
```
Comment (Alkarex) [inline Docker/entrypoint.sh:25]: Access rights need to be ensured also before our CLI commands, in the case of external volumes with incompatible permissions.

File: cli/access-permissions.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+if [ "$(id -u)" -ne 0 ]; then
+	echo >&2 '⛔ Applying access permissions require running as root or sudo!'
+	exit 3
+fi
+
+# Based on group access
+chown -R :www-data .
+# Read files, and directory traversal
+chmod -R g+rX .
```
Comment (Alkarex) [inline cli/access-permissions.sh:17]: `+X` (directory search) was what was missing in particular for some of the reported user issues

**Comment by Frenzie** [comment]
+X is only the owner?

**Comment by Alkarex** [comment]
> +X is only the owner?

`g+X` gives directory traversal permissions to the group (www-data)

**Comment by Frenzie** [comment]
I don't really understand what that means. How does that differ from recursive?

**Comment by Alkarex** [comment]
> I don't really understand what that means. How does that differ from recursive?

* `-R` (for recursive) applies the permissions to the current folder and everything below. We are doing that.
* `+X` (uppercase) is a type of permission, just like `+r` (read) and `+w` (write). It is the `+x` (lowercase) but only for directories and allows listing the content of directories (also called *search* in man page).

**Review by ColonelMoutarde — APPROVED** [review]

**Pull request merged**